httpclient/src/main/kotlin/tbdex/sdk/httpclient/TbdexHttpClient.kt

```diff
@@ -70,10 +75,11 @@ object TbdexHttpClient {
   /**
    * Sends a message to the PFI.
    *
-   * @param message The [Message] object containing the message details to be sent.
+   * @param sendMessageRequest The [SendMessageRequest] object containing the message, and optionally, a `replyTo` string for callback URL.
    * @throws TbdexResponseException for request or response errors.
    */
-  fun sendMessage(message: Message) {
+  fun sendMessage(sendMessageRequest: SendMessageRequest) {
```

Comment:
This reads like javascript. We should check at compile-time that `replyTo` is only present with RFQs. Currently, this does the check at runtime. I'm wondering if there's a more kotlin-idiomatic way to write this.

What do you think of separating the methods?
```kotlin
// Sends messages, including RFQs, but doesn't support replyTo`
sendMessage(message: Message)
// Sends only RFQs and supports replyTo
sendRfq(rfq: Rfq, replyTo: string?)
```

Or maybe overloaded methods if you want to keep the name `sendMessage` for both.
```kotlin
// Sends messages, including RFQs, but doesn't support replyTo`
sendMessage(message: Message)
// Sends only RFQs and supports replyTo
sendMessage(rfq: Rfq, replyTo: string?)
```

Reply:
i did `sendMessage(message:Message, replyTo: String? = null)`, since sendMessage() is used anytime the client is sending any tbdex message. i think the second option can also work...

Reply:
wondering what we want the dev ex to be like. should we separate it out at the method level, or should we let the client just use one method and this method takes care of the validation / inclusion of `replyTo` field in the request?

Reply:
discussed with @diehuxx - going with the second approach, with `sendMessage(rfq: Rfq, replyTo: string)` for the overloaded method.